packages/test/src/test-interface-type-safety.ts

```diff
@@ -66,3 +67,40 @@ test('Can call signal on any WorkflowHandle', async (t) => {
 
   t.pass();
 });
+
+test('Signal handler type safety', (t) => {
+  const signal = defineSignal<[string]>('a');
+
+  wf.setHandler(signal, (arg: string): void => {});
+
+  // @ts-expect-error signal handler must take string argument
+  wf.setHandler(signal, (arg: number) => {});
+
+  // @ts-expect-error signal handler must take string argument
+  wf.setHandler(signal, () => {});
+
+  // @ts-expect-error signal handler must return void
+  wf.setHandler(signal, (arg: string): string => '');
+
+  t.pass();
+});
+
+test('Query handler type safety', (t) => {
+  const query = defineQuery<string, [string]>('a');
+
+  wf.setHandler(query, (arg: string): string => '');
+
+  // @ts-expect-error query handler argument type must match
+  wf.setHandler(query, (arg: number): string => '');
+
+  // @ts-expect-error query handler argument type must match
+  wf.setHandler(query, (): string => '');
+
+  // @ts-expect-error query handler return type must match
+  wf.setHandler(query, (arg: string): void => {});
+
+  // @ts-expect-error query handler return type must match
+  wf.setHandler(query, (arg: string): number => 7);
```

Comment:
Also assert that `wf.setHandler(query, async (arg: string): Promise<string> => '')` is not accepted.